TUM-Dev/Campus-Android
# Bugfix/allow grade updates

## Issue

This fixes the following issue(s):
- Resolves: #1559

It is now taken into account that grades can be changed even after they were provided by TUM online.

## Changes
Instead of only checking whether the downloaded course is part of the existing course titles, the actual grade is also compared.

If the downloaded exam is not yet stored in the list -> store it.
If the exam was manually added and is now also available in TUM online -> remove the manually added Flag.
If the exam was not manually added and the grade changed, the grade must be updated as described in the issue.
In both of the later cases, all parameters, which can be added manually must be updated as well.
 
## How to test
Before deploying the new version:
1. Refresh grades view
2. Manually add at least one exam
3. Manually set the credits/weight for at least one exam downloaded from TUM online

Deploy this version
1. Are all the previous added/adapted exams still available and have the same parameters as before?
2. Ideally get a grade improvement in at least on course to check the functionality.
